spring-integration-redis/src/test/java/org/springframework/integration/redis/outbound/ReactiveRedisStreamMessageHandlerTests.java

```diff
@@ -0,0 +1,223 @@
+	/*@Test
+	@RedisAvailable*/
+	public void explicitJsonSerializationContextWithModelTest() {
+		String streamKey = "myStream";
+
+		RedisMessageStoreTests.Address address = new RedisMessageStoreTests.Address().withAddress("Rennes, France");
+		RedisMessageStoreTests.Person person = new RedisMessageStoreTests.Person(address, "Attoumane");
+
+		Message message = new GenericMessage(person);
+
+		RedisSerializationContext<String, Object> jsonSerializationContext = redisStringOrJsonSerializationContext(false, RedisMessageStoreTests.Person.class);
+
+		ReactiveRedisStreamMessageHandler streamMessageHandler = new ReactiveRedisStreamMessageHandler(streamKey, redisConnectionFactory);
+		streamMessageHandler.setSerializationContext(jsonSerializationContext);
+		//initializes reactiveRedisStreamOperations
+		invokeOnInitMethod(streamMessageHandler);
+
+		ReactiveMessageHandlerAdapter handlerAdapter = new ReactiveMessageHandlerAdapter(streamMessageHandler);
+		handlerAdapter.handleMessage(message);
+
+		ReactiveRedisTemplate<String, String> template = new ReactiveRedisTemplate(redisConnectionFactory, jsonSerializationContext);
+		ObjectRecord<String, RedisMessageStoreTests.Person> record = template.opsForStream().read(RedisMessageStoreTests.Person.class, StreamOffset.fromStart(streamKey)).blockFirst();
+		assertThat(record.getStream()).isEqualTo(streamKey);
+		assertThat(record.getValue().getName()).isEqualTo("Attoumane");
+		assertThat(record.getValue().getAddress().getAddress()).isEqualTo("Rennes, France");
+		template.delete(streamKey).block();
+	}
+
+
+	private RedisSerializationContext redisStringOrJsonSerializationContext(boolean string, Class jsonTargetType) {
+
+		RedisSerializationContext redisSerializationContext;
+		RedisSerializer jsonSerializer = null;
+
+		if (jsonTargetType != null) {
+			jsonSerializer = new Jackson2JsonRedisSerializer(jsonTargetType);
+		}
+		RedisSerializer stringSerializer = StringRedisSerializer.UTF_8;
+		RedisSerializationContext.SerializationPair stringSerializerPair = RedisSerializationContext.SerializationPair
+				.fromSerializer(stringSerializer);
+
+		if (string) {
+			redisSerializationContext = RedisSerializationContext
+					.newSerializationContext()
+					.key(stringSerializerPair)
+					.value(stringSerializer)
+					.hashKey(stringSerializer)
+					.hashValue(stringSerializer)
+					.build();
+		}
+		else {
+			redisSerializationContext = RedisSerializationContext
+					.newSerializationContext()
+					.key(stringSerializerPair)
+					.value(jsonSerializer)
+					.hashKey(jsonSerializer)
+					.hashValue(jsonSerializer)
+					.build();
+		}
+
+		return redisSerializationContext;
+	}
+
+	private void invokeOnInitMethod(ReactiveRedisStreamMessageHandler streamMessageHandler) {
+		try {
+			Method onInit = ReactiveRedisStreamMessageHandler.class.getDeclaredMethod("onInit");
+			onInit.setAccessible(true);
+			onInit.invoke(streamMessageHandler);
+		}
+		catch (Exception e) {
+			e.printStackTrace();
```

Comment:
That's not good. you should re-throw an exception.
Although you don't need such a reflection exercise because you should just call `public afterPropertiesSet()`.

I also don't see that you populate an `ApplicationContext` and `BeanFactory` to components you create manually...